pkg/status/combinedstatus-resolution.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package status
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+
+	"github.com/kubestellar/kubestellar/api/control/v1alpha1"
+	"github.com/kubestellar/kubestellar/pkg/util"
+)
+
+// combinedStatusResolution is a struct that represents the resolution of a
+// combinedstatus. A combinedstatus resolution maps to (binding, object).
+// A resolution's logic is as follows:
+// - Composition:
+//   - A combinedstatus resolution is composed of multiple statuscollector
+//     data.
+//   - statuscollector data is composed of collector-clauses. Each
+//     collector-clause has an expression that is evaluated against every
+//     workstatus that is associated with the statuscollector, and the result
+//     is cached.
+//
+// - Flow:
+//   - When a statuscollector is added to the combinedstatus resolution,
+//     every clause in the statuscollector is evaluated against every workstatus
+//     that is associated with the statuscollector.
+//   - When a workstatus is added to / modified in the combinedstatus
+//     resolution, every clause in every statuscollector is evaluated against
+//     the workstatus. The caching of collector-clause -> evaluation makes
+//     calculating combinedFields efficient if present.
+//
+// - Ownership:
+//   - A combinedstatus resolution is owned by the binding that it is
```

Comment:
"owned by" is a bit vague, can you say what is meant by that here?

Reply:
removed